src/main/java/com/stuypulse/robot/util/Encoder.java

```diff
@@ -0,0 +1,99 @@
+package com.stuypulse.robot.util;
+
+import com.revrobotics.CANEncoder;
+import com.stuypulse.robot.subsystems.Drivetrain.Gear;
+
+/**
+ * A wrapper class for a 'raw' encoder that provides 
+ * additional functionality like handling gear shifting and
+ * can make the encoder easier to use / the methods easier to 
+ * understand. 
+ * 
+ * TODO: velocity needs to be tested
+ * 
+ * @author Myles Pasetsky (@selym3)
+ */
+public class Encoder {
+    
+    // The encoder class that is closest to hardware
+    private CANEncoder encoder;
+
+    // These values are used to integrate an accurate distance
+    // while switching gears 
+    private double accumulated;
+    private double lastRotations;
+
+    /**
+     * Create an Encoder wrapper
+     * 
+     * @param encoder the underlying encoder class
+     */
+    public Encoder(CANEncoder encoder) {
+        this.encoder = encoder; 
+        reset();
+    }
+
+    /**
+     * Get the raw encoder rotations (the unscaled value)
+     * 
+     * @return encoder rotations
+     */
+    public double getRotations() {
+        return encoder.getPosition();
+    }
+
+    /**
+     * Gets the distance the encoder has traveled (the scaled value)
+     * 
+     * @return distance traveled
+     */
+    public double getDistance() {
+        return accumulated;
+    }
+    
+    /**
+     * Gets the raw rotations per minute (the unscaled value)
+     * 
+     * @return rotations per minute
+     */
+    public double getRPM() {
+        return encoder.getVelocity();
```

Comment:
this is not in RPM, it is the derivative of `encoder.getDistance()`, if it is scaled by the gear then it will be accurate.